api/exec/compose_stack.go

```diff
@@ -115,3 +123,129 @@ func createEnvFile(stack *portainer.Stack) (string, error) {
 
 	return "stack.env", nil
 }
+
+func createNetworkEnvFile(stack *portainer.Stack) error {
+	ss := NewStringSet()
+
+	for _, filePath := range stackutils.GetStackFilePaths(stack) {
+		ssName, err := extractNetworkName(filePath)
+		if err != nil {
+			return err
+		}
+
+		if ssName == nil || ssName.Len() == 0 {
+			continue
+		}
+
+		ss.Merge(ssName)
+	}
+
+	for _, s := range ss.List() {
+		if _, ok := os.LookupEnv(s); ok {
+			ss.Remove(s)
+		}
+	}
+
+	if ss.Len() == 0 {
+		return nil
+	}
+
+	envfile, err := os.OpenFile(path.Join(stack.ProjectPath, ".env"),
+		os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+
+	defer envfile.Close()
+
+	var stackFunc = func(name string) (string, bool) {
+		if stack.Env != nil {
+			for _, v := range stack.Env {
+				if name == v.Name {
+					return v.Value, true
+				}
+			}
+		}
+
+		return "", false
+	}
+
+	for _, s := range ss.List() {
+		if v, ok := stackFunc(s); ok {
+			envfile.WriteString(fmt.Sprintf("%s=%s\n", s, v))
+		} else {
+			envfile.WriteString(fmt.Sprintf("%s\n", s))
+		}
+	}
+
+	return nil
+}
+
+func extractNetworkName(filePath string) (StringSet, error) {
```

Comment:
are we looking for the stack network names?
```suggestion
func extractNetworkNames(filePath string) (StringSet, error) {
```

Reply:
Sure